app/src/main/java/org/p2p/wallet/home/ui/main/empty/EmptyViewAdapter.kt

```diff
@@ -0,0 +1,43 @@
+package org.p2p.wallet.home.ui.main.empty
+
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import org.p2p.wallet.R
+import org.p2p.wallet.home.model.HomeBanner
+import org.p2p.wallet.home.model.HomeElementItem
+import org.p2p.wallet.home.ui.main.adapter.OnHomeItemsClickListener
+
+class EmptyViewAdapter(
+    private val listener: OnHomeItemsClickListener
+) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+
+    private val data = mutableListOf<Any>()
+
+    private var isZerosHidden: Boolean = true
+
+    fun setItems(items: List<Any>) {
+        data.clear()
+        data.addAll(items)
+        notifyDataSetChanged()
+    }
+
+    override fun getItemViewType(position: Int): Int = when (data[position]) {
+        is HomeBanner -> R.layout.item_big_banner
+        else -> throw IllegalStateException("Unknown viewType: ${data[position]}")
+    }
+
+    override fun getItemCount(): Int = data.size
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder = when (viewType) {
+        R.layout.item_big_banner -> BigBannerViewHolder(parent) { bannerId: Int -> listener.onBannerClicked(bannerId) }
+        R.layout.item_get_token -> GetTokenViewHolder(parent, listener)
```

Comment:
i would make BigBannerViewHolder and GetTokenViewHolder children of sealed class